packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -477,25 +494,29 @@ class _CupertinoDatePickerDateTimeState extends State<CupertinoDatePicker> {
         if (widget.use24hFormat) {
           selectedHour = index;
           widget.onDateTimeChanged(_getDateTime());
-        }
-        else {
-          final int currentHourIn24h = selectedHour + selectedAmPm * 12;
+        } else {
+          selectedHour = index % 12;
+
           // Automatically scrolls the am/pm column when the hour column value
-          // goes far enough. This behavior is similar to
-          // iOS picker version.
-          if (currentHourIn24h ~/ 12 != index ~/ 12) {
-            selectedHour = index % 12;
-            amPmController.animateToItem(
-              1 - amPmController.selectedItem,
-              duration: const Duration(milliseconds: 300), // Set by comparing with iOS version.
-              curve: Curves.easeOut,
-            ); // Set by comparing with iOS version.
+          // goes far enough.
+
+          if (
+            ((previousHourIndex >= 0 && previousHourIndex <= 11) && (index >= 12 && index <= 23)) ||
```

Comment:
Can you extract the various inner conditions to named variables such that it's then

```dart
(previouslyAm && nowPm) || (previouslyPm && nowAm)
```
or some such

Reply:
I thought of a way to simplify it further based on your suggestion, what do you think?